azure_functions_worker/bindings/meta.py

```diff
@@ -120,13 +128,16 @@ def to_outgoing_proto(binding: str, obj: typing.Any, *,
 def to_outgoing_param_binding(binding: str, obj: typing.Any, *,
                               pytype: typing.Optional[type],
                               out_name: str,
-                              shmem_mgr) \
+                              shmem_mgr,
+                              is_function_data_cache_enabled: bool) \
         -> protos.ParameterBinding:
     datum = get_datum(binding, obj, pytype)
     shared_mem_value = None
     # If shared memory is enabled and supported for the given datum, try to
     # transfer to host over shared memory as a default
-    if shmem_mgr.is_enabled() and shmem_mgr.is_supported(datum):
+    can_transfer_over_shmem = shmem_mgr.is_supported(datum) or \
```

Comment:
Let's put this whole thing condition in a function?

Reply:
The condition for this could still be made more readable.
I think adding all in a single function is cleaner.
```
def function_name(XXXX):
  if shmem_mgr.is_enabled():
    return False
  if not is_function_data_cache_enabled:
    return False
  if not _does_datatype_support_caching(datum):
    return False
  if not shmem_mgr.is_supported(datum):
    return False
  return True
```